main.go

```diff
@@ -111,8 +116,15 @@ func run(c *cli.Context) error {
 		go func() { startchan <- struct{}{} }()
 		file := c.String("file")
 		configchan := make(chan bool)
+		// XXX Unwatchable Puppet sources should get events from a timer,
+		// with the same interval as the Puppet agent
+		// (hint: puppet config print runinterval)
 		if !c.Bool("no-watch") {
```

Comment:
This code isn't correct, because it's wrapped inside the no-watch flag. Instead, I think it's more appropriate for the puppet thing to have it's own first class entry in the select loop just below this... Reminder, a nil channel is always ignored, so if we're not using this, then a nil value means we never run that case statement.